hivemind/dht/node.py

```diff
@@ -289,108 +319,198 @@ async def get(self, key: DHTKey, latest=False, **kwargs) -> Tuple[Optional[DHTVa
         Search for a key across DHT and return either first or latest entry.
         :param key: same key as in node.store(...)
         :param latest: if True, finds the latest value, otherwise finds any non-expired value (which is much faster)
-        :param kwargs: parameters forwarded to get_many
+        :param kwargs: parameters forwarded to get_many_by_id
         :returns: (value, expiration time); if value was not found, returns (None, None)
         """
         if latest:
             kwargs["sufficient_expiration_time"] = float('inf')
         result = await self.get_many([key])
         return result[key]
 
-    async def get_many(
-            self, keys: Collection[DHTKey], sufficient_expiration_time: Optional[DHTExpiration] = None,
-            num_workers: Optional[int] = None, beam_size: Optional[int] = None
-    ) -> Dict[DHTKey, Tuple[Optional[DHTValue], Optional[DHTExpiration]]]:
+    async def get_many(self, keys: Collection[DHTKey], sufficient_expiration_time: Optional[DHTExpiration] = None,
+                       **kwargs) -> Dict[DHTKey, Union[Tuple[Optional[DHTValue], Optional[DHTExpiration]],
```

Comment:
Side note: the type of return value probably means that we're doing something wrong... We should at least think of separating the logic in order to conform to the SRP